streams/src/test/java/org/apache/kafka/streams/processor/internals/StateDirectoryTest.java

```diff
@@ -77,6 +84,7 @@
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
+//TODO KAFKA-12648: add tests for named topology specific stuff
```

Comment:
I've added a few tests for the named topology stuff but I definitely want to add more and just haven't yet had time. Since I'm still on-call and therefore unlikely to have time until next week, if you both are able to do a quick pass and don't have any further feedback on the PR as-is, it may make sense to just merge this PR tomorrow (Friday) and do a quick followup PR for the tests next week.

That way  I can rebase the next PR  ([Pt. 2](https://github.com/apache/kafka/pull/10683)) and you all can actually begin reviewing that. cc @guozhangwang @wcarlson5